verilog/tools/kythe/indexing_facts_tree_extractor_test.cc

```diff
@@ -82,101 +88,424 @@ TEST(ExtractOneFileTest, FactsTreeExtractor) {
-  constexpr absl::string_view code_text = "";
+TEST(EmptyModuleTest, FactsTreeExtractor) {
+  constexpr int kTag = 1;  // value doesn't matter
+  const verible::SyntaxTreeSearchTestCase kTestCase = {
+      {"module ", {kTag, "foo"}, "; endmodule: ", {kTag, "foo"}}};
+
   constexpr absl::string_view file_name = "verilog.v";
   int exit_status = 0;
   bool parse_ok = false;
 
-  const auto expected =
-      T({{Anchor(file_name, 0, code_text.size())}, IndexingFactType ::kFile});
+  const auto expected = T(
+      {
+          {
+              Anchor(file_name, 0, kTestCase.code.size()),
+              Anchor(kTestCase.code, 0, kTestCase.code.size()),
+          },
+          IndexingFactType ::kFile,
+      },
+      T({
+          {
+              Anchor(kTestCase.expected_tokens[1], kTestCase.code),
+              Anchor(kTestCase.expected_tokens[3], kTestCase.code),
+          },
+          IndexingFactType::kModule,
+      }));
 
   const auto facts_tree =
-      ExtractOneFile(code_text, file_name, exit_status, parse_ok);
+      ExtractOneFile(kTestCase.code, file_name, exit_status, parse_ok);
 
   const auto result_pair = DeepEqual(facts_tree, expected);
   EXPECT_EQ(result_pair.left, nullptr) << *result_pair.left;
   EXPECT_EQ(result_pair.right, nullptr) << *result_pair.right;
 }
 
-TEST(EmptyModuleTest, FactsTreeExtractor) {
-  constexpr absl::string_view code_text = "module foo; endmodule: foo";
+TEST(OneModuleInstanceTest, FactsTreeExtractor) {
+  constexpr int kTag = 1;  // value doesn't matter
+  const verible::SyntaxTreeSearchTestCase kTestCase = {{"module ",
+                                                        {kTag, "bar"},
+                                                        "; endmodule: ",
+                                                        {kTag, "bar"},
+                                                        " module ",
+                                                        {kTag, "foo"},
+                                                        "; ",
+                                                        {kTag, "bar"},
+                                                        " ",
+                                                        {kTag, "b1"},
+                                                        "(); endmodule: ",
+                                                        {kTag, "foo"}}};
+
   constexpr absl::string_view file_name = "verilog.v";
   int exit_status = 0;
   bool parse_ok = false;
 
-  const auto expected =
-      T({{Anchor(file_name, 0, code_text.size())}, IndexingFactType ::kFile},
-        T({{Anchor(absl::string_view("foo"), 7, 10),
-            Anchor(absl::string_view("foo"), 23, 26)},
-           IndexingFactType::kModule}));
+  const auto expected = T(
+      {
+          {
+              Anchor(file_name, 0, kTestCase.code.size()),
+              Anchor(kTestCase.code, 0, kTestCase.code.size()),
+          },
+          IndexingFactType ::kFile,
+      },
+      T({
```

Comment:
As the test cases get bigger, add a comment near each node describing what is being referenced., e.g.

`// refers to module bar`